server/repository/src/db_diesel/temperature_chart/temperature_chart.rs

```diff
@@ -0,0 +1,355 @@
+use crate::{
+    db_diesel::temperature_log_row::temperature_log::dsl as temperature_log_dsl, DBType,
+    RepositoryError, StorageConnection, TemperatureChartRow, TemperatureLogFilter,
+    TemperatureLogRepository,
+};
+use chrono::NaiveDateTime;
+use diesel::prelude::*;
+use util::datetime_with_offset;
+
+use super::temperature_chart_row::*;
+
+pub struct TemperatureChartRepository<'a> {
+    connection: &'a StorageConnection,
+}
+
+type QueryResult = (NaiveDateTime, NaiveDateTime, f64, String);
+
+impl<'a> TemperatureChartRepository<'a> {
+    pub fn new(connection: &'a StorageConnection) -> Self {
+        TemperatureChartRepository { connection }
+    }
+
+    pub fn query(
+        &self,
+        from_datetime: NaiveDateTime,
+        to_datetime: NaiveDateTime,
+        number_of_data_points: i32,
+        temperature_log_filter: Option<TemperatureLogFilter>,
+    ) -> Result<(Vec<TemperatureChartRow>, Vec<Interval>), RepositoryError> {
```

Comment:
Intervals returned with temperature chart, so that full chart can be drawn without gaps (should be easy to map to recharts)

Reply:
In the returned result, there's no need for the `to_datetime`
To map values on a chart, we need

```
{
  datetime: NaiveDatetime,
  temperature: Double,
  sensorId: String,
  breachId: String
}
```

really, it's just the datetime and temperature, but the sensor and breach are also needed if we are to show a breach indicator & popup which has location and other sensor details.

Reply:
> In the returned result, there's no need for the to_datetime

Oki, should the datetime return by API be the middlepoint between from_datetime and to_datetime ?

>  but the sensor and breach are also needed if we are to show a breach indicator & popup which has location and other sensor details.

Sounds good, I think we can return an array of breachIds, change to follow